src/AutoRest.CSharp/MgmtTest/AutoRest/SourceCodeProject.cs

```diff
@@ -41,9 +42,11 @@ public async Task<Compilation> GetCompilationAsync()
         private static Project CreateSourceCodeProject(string sourceCodePath, string[] sharedSourceFolders)
         {
             var sourceCodeProject = CreateGeneratedCodeProject();
-            var sourceCodeGeneratedDirectory = Path.Join(sourceCodePath, GeneratedCodeWorkspace.GeneratedFolder);
+            DirectoryInfo di = new DirectoryInfo(sourceCodePath);
+            var genFolders = di.EnumerateDirectories(GeneratedCodeWorkspace.GeneratedFolder, SearchOption.AllDirectories).ToList();
 
-            sourceCodeProject = GeneratedCodeWorkspace.AddDirectory(sourceCodeProject, sourceCodePath, skipPredicate: sourceFile => sourceFile.StartsWith(sourceCodeGeneratedDirectory));
+            sourceCodeProject = GeneratedCodeWorkspace.AddDirectory(sourceCodeProject, sourceCodePath,
+                skipPredicate: sourceFile => genFolders.Exists(f => sourceFile.StartsWith(f.FullName)));
```

Comment:
Could we add some comments to describe the things going on here?

Reply:
sure